pyfgaws/batch/api.py

```diff
@@ -391,6 +414,19 @@ def wait_on(
         model: botocore.waiter.WaiterModel = botocore.waiter.WaiterModel(config)
         waiter: BotoWaiter = botocore.waiter.create_waiter_with_client(name, model, self.client)
         waiter.wait(jobs=[self.job_id])
+
+        if self.cancel_on is not None:
+            for code in self.cancel_on:
+                signal.signal(
+                    code, lambda signum, frame: self.cancel_job(reason=f"Interrupted: {code}")
+                )
+
+        if self.terminate_on is not None:
+            for code in self.terminate_on:
+                signal.signal(
+                    code, lambda signum, frame: self.terminate_job(reason=f"Interrupted: {code}")
+                )
```

Comment:
this needs a unit test probably.  Make a sub-process, then check that it terminates?